iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift

```diff
@@ -18,10 +18,27 @@ extension ConversationListViewController {
             }
         }
 
-        self.dataSource.handleSelectedMessage = { [unowned self] (message) in
-            guard let cid = try? ConversationID(cid: message.conversationId) else { return }
+        self.dataSource.handleSelectedMessage = { [unowned self] (conversation) in
+            guard let cid = try? ConversationID(cid: conversation.conversationId) else { return }
             self.onSelectedConversation?(cid)
         }
+
+        self.dataSource.handleDeleteMessage = { (conversation) in
+            Task {
+                do {
+                    let cid = try ConversationID(cid: conversation.conversationId)
+                    let conversationController = ChatClient.shared.channelController(for: cid)
+                    if conversationController.conversation.isFromCurrentUser {
+                        try await conversationController.deleteChannel()
+                    } else {
+                        try await conversationController.hideChannel()
+                    }
```

Comment:
Delete or hide the conversation based on permissions. Hiding means the conversation won't be shown until someone else sends a message to it.

Reply:
Does this need be added to a task pool?